File: 알고리즘/Algorithm_JAVA/src/Bubble.java
public class Bubble {
    public void bubbleSort(int[] data) {
        int i,j,temp;
        int size = data.length;

        for(i=0; i < size; i++) {
            for(j=0; j < size - i - 1; j++) {
                if(data[j] > data[j+1]) {
                    temp = data[j+1];
                    data[j+1] = data[j];
                    data[j] = temp;
                }
            }
        }

        for(int k = 0; k < size; k++) {
            System.out.print(data[k] + " ");
        }
    }
}

File: Algorithm/Algorithm_C++/Algorithm/bubble.cpp
#include <iostream>

using namespace std;

void bubbleSort(int arr[], int len) {
	int i, j, temp;

	for (i = 0; i < len; i++) {
		for (j = 0; j < len - (i + 1); j++) {
			if (arr[j] > arr[j + 1]) {
				temp = arr[j + 1];
				arr[j + 1] = arr[j];
				arr[j] = temp;
			}
		}
	}
}

int main() {
	int arr[6] = { 5,3,8,1,2,7 };
	int len = (sizeof(arr) / sizeof(int));

	cout << "정렬 전: ";
	for (int x : arr)
		cout << x << " ";
	cout << endl;

	bubbleSort(arr, len);

	cout << "정렬 후: ";
	for (int x : arr)
		cout << x << " ";
}